unit/java_bytecode/java_bytecode_parse_generics/WildcardGenericFunctions.java

```diff
@@ -1,54 +1,43 @@
 public class WildcardGenericFunctions
 {
   // Test a wild card generic type
-  public static void processSimpleGeneric(SimpleGeneric<?> x) {
-    assert(x.t.equals(null));
+  public static void processSimpleGeneric(Generic<?> x)
+  {
+
   }
 
   // Test a wildcard generic bound by an interface
-  public static void processUpperBoundInterfaceGeneric(SimpleGeneric<? extends BasicInterface> x) {
-    assert(x.t.getX() == 4);
+  public static void processUpperBoundInterfaceGeneric(Generic<? extends
+  Interface> x)
+  {
+
   }
 
   // Test a wild card generic bound by a class
-  public static void processUpperBoundClassGeneric(SimpleGeneric<? extends Foo> x) {
-    assert(x.t.getX() == 4);
+  public static void processUpperBoundClassGeneric(Generic<? extends
+  Interface_Implementation> x)
+  {
+
   }
 
-  // It isn't legal to have an wild card with two upper bounds
+  // It isn't legal to have a wild card with two upper bounds
   // Per language spec on intersection types
 
-  public static void processLowerBoundGeneric(SimpleGeneric<? super Foo> x, Foo assign) {
-    x.t = assign;
+  public static void processLowerBoundGeneric(Generic<? super
+  Interface_Implementation> x, Interface_Implementation assign)
+  {
+
   }
 
   // It is not legal Java to specify both an upper and lower bound
-  // public static void processBoundSuperClassGeneric(SimpleGeneric<? extends Object super Foo> x, Foo assign) {
-  //   x.t = assign;
+  // public static void processBoundSuperClassGeneric(Generic<? extends Object super Interface_Implementation> x, Interface_Implementation assign)
```

Comment:
These commented out parts are presumably not expected to be supported yet? While we may not have unit tests to support these features yet, is it worth still having this code in the .java/.class files anyway to at least test/prove that we at least don't crash when encountering them? Alternatively, if we *do* currently crash when encountering them, is there a Jira issue covering that crash, and if so, it would be nice if that Jira issue was reference from within these comments, e.g. ` // This currently causes a crash, but should be re-enabled onse TG-1234 is fixed` - having comments like that helps make it easier to re-enable useful tests once things get fixed, or to at least understand why something is commented out :-)

Reply:
As Thomas pointed out, this is not valid Java code but it may be a good idea to leave it here, in case someone would want to extend the test and would ponder about the same example ;) The only thing that changed here really is the renaming and removing the assignment.